### PR TITLE
pipeline

### PR DESCRIPTION
**Pull request by szympie**

### CODE REVIEW

**Review by khrynczenko — COMMENTED** [review]
Are we capable of loading all images at once and storing them in array? I'm not sure, seems like a lot to store. I think we need to process them one by one.

File: Utilities/Pipeline/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+
+class PipelineDataPoint(object):
+    """
+    Basic data type in the pipeline, it is used as a container
+    to pass data between strategies.
+    """
+
+    def __init__(self, data=None, id=None):
+        if data is None and id is None:
```
Comment (mmyller) [inline Utilities/Pipeline/pipeline.py]: Will this trigger when I pass the id but don't pass the data?

**Review by mmyller — APPROVED** [review]

**Comment by khrynczenko** [comment]
I would like to see some tests.

**Review by khrynczenko — APPROVED** [review]

File: Utilities/Pipeline/pipeline.py (hunk 1 above)
Comment (szympie) [reply Utilities/Pipeline/pipeline.py]: Fixed, both values should be passed.

**Review by mmyller — APPROVED** [review]

**Pull request merged**